lint/rule_target_required_matchers.go

```diff
@@ -0,0 +1,91 @@
+package lint
+
+import (
+	"fmt"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type TargetRequiredMatchersRuleSettings struct {
+	Matchers config.Matchers `yaml:"matchers"`
+}
+
+func NewTargetRequiredMatchersRule(config *TargetRequiredMatchersRuleSettings) *TargetRuleFunc {
+	return &TargetRuleFunc{
+		name:        "target-required-matchers-rule",
+		description: "Checks that target PromQL query has the required matchers",
+		stability:   "experimental",
+		fn: func(d Dashboard, p Panel, t Target) TargetRuleResults {
+			r := TargetRuleResults{}
+			// TODO: The RuleSet should be responsible for routing rule checks based on their query type (prometheus, loki, mysql, etc)
+			// and for ensuring that the datasource is set.
+			if t := getTemplateDatasource(d); t == nil || t.Query != Prometheus {
+				// Missing template datasource is a separate rule.
+				// Non prometheus datasources don't have rules yet
+				return r
+			}
+
+			expr, err := parsePromQL(t.Expr, d.Templating.List)
+			if err != nil {
+				// Invalid PromQL is another rule
+				return r
+			}
+			for _, m := range config.Matchers {
```

Comment:
Could config be nil here in the case that the rule is not configured? Probably need a check.